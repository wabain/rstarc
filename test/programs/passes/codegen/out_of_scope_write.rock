Put nothing into the sun
Perseverance takes a lot
The sun is like a smooth meadow

Say the sun
