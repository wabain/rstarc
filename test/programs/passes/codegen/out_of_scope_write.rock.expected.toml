[tests]
enabled = ["pretty", "ir", "run"]

[pretty]
returncode = "0"
stdout = """
Put null into the sun
Perseverance takes a lot
Put 4166 into the sun

Say the sun
"""

[ir]
returncode = "0"
stdout = """
global the sun<0>

main:
  Perseverance := store mysterious
  the sun := store null
  Perseverance := store Function(1)
  say the sun
  return-default

;
; Function Perseverance takes a lot
; Scope is 1
;
f1_Perseverance:
  a lot := load-arg 0
  the sun<0> := store 4166
  return-default
"""

[run]
returncode = "0"
stdout = """
null
"""
